app/services/create_newsletter_signup.rb

```diff
@@ -2,23 +2,31 @@
 include SendGrid
 
 class CreateNewsletterSignup
-  attr_accessor :first_name, :last_name, :email, :list_ids, :api_key
+  attr_accessor :first_name, :last_name, :email, :status, :list_ids, :api_key
 
-  def initialize(first_name:, last_name:, email:, api_key: ENV['SENDGRID_API_KEY'], list_ids: [ENV['NEWSLETTER_LIST_ID']])
+  def initialize(first_name:, last_name:, email:, status:, api_key: ENV['SENDGRID_API_KEY'], list_ids: [ENV['NEWSLETTER_LIST_ID']])
     @first_name = first_name
     @last_name = last_name
     @email = email
+    @status = status
     @api_key = api_key
     @list_ids = list_ids
   end
 
   def perform
-    signup_api_call
+    call = sendgrid_api_call
+    if call
+      'success'
+    else
+      'error'
+    end
+  rescue => e
+    'error'
   end
```

Comment:
Calling `perform` on this service returns just a string... what if this were where you return a result object?